storage/leveldb/batch.go

```diff
@@ -1,35 +1,60 @@
 package leveldb
 
 import (
+	"sync"
+
 	"github.com/syndtr/goleveldb/leveldb"
 )
 
+const removed = "removed"
+
 type batch struct {
-	batch *leveldb.Batch
+	batch         *leveldb.Batch
+	cachedData    map[string][]byte
+	mutCachedData sync.RWMutex
```

Comment:
since this mutex also protects against the simultaneous batch writes (which is very good), might rename it to mutBatch

Reply:
done